hub/auto/unstructured/image_classification.py

```diff
@@ -122,9 +123,11 @@ def structure(  # type: ignore
 
         with ds:
```

Comment:
```suggestion
        with ds, iterator:
```

if you move the iterator outside this with statement (above) you can use this syntax, and i think you can remove the `iterator.close()` call after all the loops

Reply:
test this first though, im not 100% sure

Reply:
`iterator.close()` isn't necessary there. I'll remove it